test/set_test.js

```diff
@@ -0,0 +1,153 @@
+goog.module('datalayerhelper.helper.testing.set');
+goog.setTestOnly();
+
+const {DataLayerHelper} = goog.require('helper');
+
+describe('The set command', function() {
+  let dataLayer;
+  let dlh;
+  let commandAPI;
+  beforeAll(function() {
+    dataLayer = [];
+    commandAPI = function() {
+      dataLayer.push(arguments);
+    };
+    dlh = new DataLayerHelper(dataLayer);
+    commandAPI('set', 'bar', 'foo');
+  });
+
+  describe('with 2 argument format', function() {
+    it('updates model without corrupting it.', function() {
+      const targetModel = Object.assign({}, dlh.abstractModelInterface_);
```

Comment:
Some of these tests could be cleaned up a bit. I think it would be better to use "jasmine.objectContaining". This test would look as follows:

```
it('updates model without corrupting it.', () => {
  commandAPI('set', 'foo', 'bar');
  expect(dlh.abstractModelInterface_).toEqual(jasmine.objectContaining({'foo': 'bar'});
});
```

Thoughts?

Reply:
I agree that this is much cleaner, but my thought process in setting the target model each time was to assert that not only was the right value being set, but all of the other values were not corrupted. Let me know what you think and I will change it if that isn't too important!

Reply:
I suppose that's fine, but personally I'd prefer to check that in a single test (i.e. by updating the model appropriately in the test set-up, executing your test, then verifying that the model is not corrupted in that specific tests).